File: backend/Models/EmployeeInvite.cs
using System.ComponentModel.DataAnnotations;

namespace backend.Models;

public class EmployeeInvite
{
    [Key]
    public Guid Id { get; set; }
    public Guid WorkspaceId { get; set; }
    public Workspace? Workspace { get; set; }
    public string Email { get; set; } = string.Empty;
    public string Token { get; set; } = string.Empty;
    public UserRole Role { get; set; } = UserRole.Employee;
    public bool IsAccepted { get; set; } = false;
    public DateTime ExpiresAt { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/src/types/EmployeeInvite.ts
import { UserRole } from '../constants/Enumeration/UserEnum/UserEnum';
import { TWorkspace } from './Workspace';

export type TEmployeeInvite = {
  id: string;
  workspaceId: string;
  workspace?: TWorkspace;
  email: string;
  token: string;
  role: UserRole;
  isAccepted: boolean;
  expiresAt: string;
  createdAt: string;
  updatedAt: string;
};

export type TEmployeeInviteRequest = {
  emails: string[];
  workspaceId: string;
};
